src/integrations/GA4/browser.js

```diff
@@ -54,10 +56,43 @@ export default class GA4 {
       gtagParameterObject.user_id = this.analytics.getUserId();
     }
 
-    gtagParameterObject.cookie_prefix = 'rs';
     if (this.isHybridModeEnabled && this.overrideClientAndSessionId) {
+      gtagParameterObject.cookie_prefix = 'rs';
       gtagParameterObject.client_id = this.analytics.getAnonymousId();
       gtagParameterObject.session_id = this.analytics.getSessionId();
+    } else {
+      // Cookie migration logic
+      const clientCookie = this.cookie.get('rs_ga');
+      const defaultGA4Cookie = this.cookie.get('_ga');
+      const measurementIdArray = this.measurementId.split('-');
+      const sessionCookie = this.cookie.get(`rs_ga_${measurementIdArray[1]}`);
+
+      // Only override when both cookie with rs prefix is available and default cookie is not available
+      if (!defaultGA4Cookie && clientCookie && sessionCookie) {
+        const clientCookieArray = clientCookie.split('.');
+
+        // client_id cookie : GA1.1.51545857.1686555747
+        // client_id cookie : GA1.1.48512441-5e44-4860-9900-a8f6e6bc4041
+        const clientId = clientCookieArray.length > 3
+          ? `${clientCookieArray[2]}.${clientCookieArray[3]}`
+          : clientCookieArray[2];
+
+        // session_id cookie : GS1.1.1686558743.1.1.1686558965.7.0.0
+        const sessionCookieArray = sessionCookie.split('.');
+        const sessionId = sessionCookieArray[2];
+
```

Comment:
Can't we use `analytics.getAnonymousId()` and `analytics.getSessionId()` for this?

Reply:
what if the session_id is changed from rs end and it's not changes from ga4 end ?
as someone wight have set custom session expire time in ga4
batter to reference cookies